    <?xml version="1.0" encoding="UTF-8"?>
    <ui version="4.0">
     <class>ExtractBuildingHeightDialogBase</class>
     <widget class="QDialog" name="ExtractBuildingHeightDialogBase">
      <property name="windowTitle"><string>Extraer altura de edificios</string></property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex"><number>0</number></property>
         <widget class="QWidget" name="tabInputs">
          <attribute name="title"><string>Parámetros</string></attribute>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0"><widget class="QLabel"><property name="text"><string>Edificios (polígono)</string></property></widget></item>
           <item row="0" column="1"><widget class="QComboBox" name="cmbBuildings"/></item>
           <item row="1" column="0"><widget class="QLabel"><property name="text"><string>Nube de puntos</string></property></widget></item>
           <item row="1" column="1"><widget class="QComboBox" name="cmbPointCloud"/></item>
           <item row="2" column="0"><widget class="QLabel"><property name="text"><string>Filtro de suelo (expresión)</string></property></widget></item>
           <item row="2" column="1"><widget class="QLineEdit" name="leGroundExpr"><property name="text"><string>Classification = 2</string></property></widget></item>
           <item row="3" column="0"><widget class="QLabel"><property name="text"><string>Resolución (m)</string></property></widget></item>
           <item row="3" column="1"><widget class="QDoubleSpinBox" name="spRes"><property name="decimals"><number>2</number></property><property name="minimum"><double>0.05</double></property><property name="value"><double>1.0</double></property></widget></item>
           <item row="4" column="0"><widget class="QLabel"><property name="text"><string>Tamaño de tesela (px)</string></property></widget></item>
           <item row="4" column="1"><widget class="QSpinBox" name="spTile"><property name="minimum"><number>100</number></property><property name="maximum"><number>10000</number></property><property name="value"><number>1000</number></property></widget></item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabHelp">
          <attribute name="title"><string>Ayuda</string></attribute>
          <layout class="QVBoxLayout" name="verticalLayoutHelp">
           <item>
            <widget class="QTextBrowser" name="helpBrowser">
             <property name="html">
              <string>
<![CDATA[
<h3>ExtractBuildingHeight — Ayuda</h3>
<ol>
<li><b>DEM</b>: ráster de elevación a partir de <i>todos</i> los puntos.</li>
<li><b>DTM</b>: ráster de terreno con filtro de suelo (p.ej. <code>Classification = 2</code>).</li>
<li><b>nDSM</b> = DEM − DTM (alturas normalizadas).</li>
<li><b>Estadística zonal</b>: máximo por polígono.</li>
<li><b>Field calc</b>: crea <code>height_rounded</code> = redondeo del máximo.</li>
</ol>
<p><b>Parámetros:</b></p>
<ul>
<li><i>Buildings</i>: capa de polígonos de edificios.</li>
<li><i>Point Cloud</i>: capa de nube de puntos (LAS/LAZ).</li>
<li><i>Ground filter</i>: expresión sobre atributos LiDAR (ej. <code>Classification = 2</code>).</li>
<li><i>Resolution</i> (m) y <i>Tile size</i> (px): controlan detalle y rendimiento de la triangulación.</li>
</ul>
<p>Requisitos: PDAL instalado y proveedor QGIS-Processing PDAL habilitado.</p>
]]>
              </string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="wButtons">
         <layout class="QHBoxLayout" name="hLayout">
          <item><widget class="QPushButton" name="btnOpenLog"><property name="text"><string>Abrir log</string></property></widget></item>
          <item><spacer name="spacer"><property name="orientation"><enum>Qt::Horizontal</enum></property></spacer></item>
          <item><widget class="QDialogButtonBox" name="buttonBox"><property name="standardButtons"><set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set></property></widget></item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <resources/>
     <connections/>
    </ui>
